model/provider.go

```diff
@@ -0,0 +1,22 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import "context"
+
+// Provider is an interface for securely loading secrets based on environment variables.
+type Provider interface {
+	LoadSecrets(ctx context.Context, envs map[string]string) ([]string, error)
```

Comment:
I thought we agreed we drop the concept of "envs". We want to get a list of secret values from the provider. The logic that replaces values in environment variables is generic, independent of providers.